### PR TITLE
Add support for XP-Pen Artist 16

### PR DESCRIPTION
**Pull request by Mat931**

### CODE REVIEW

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    {
+      "VendorID": 21827,
+      "ProductID": 77,
+      "InputReportLength": 8,
+      "OutputReportLength": 0,
+      "ReportParser": "OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenOffsetAuxReportParser",
+      "FeatureInitReport": null,
+      "OutputInitReport": null,
+      "DeviceStrings": {},
+      "InitializationStrings": [
```
Comment (jamesbt365) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:30]: Are you sure you need *every* string here for full functionality?

Judging from other tablets only 100 should be required, 123 may be required for aux functionality, could you provide a string dump from the inbuilt device string reader?

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "Name": "XP-Pen Artist 16",
+  "Specifications": {
+    "Digitizer": {
+      "Width": 340.99,
+      "Height": 191.81,
+      "MaxX": 34416,
```
Comment (jamesbt365) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:7]: Are you sure these are right? If these are the highest values you can get you should adjust the width and weight to reflect these properly.

max value/lpmm = physical size

the lpmm for this tablet is 100 and the max value is what you can measure in the `tablet > tablet debugger` when moving your pen as far as you can get it to read.

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      "MaxY": 19359
+    },
+    "Pen": {
+      "MaxPressure": 2047,
```
Comment (jamesbt365) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:11]: XP-Pen likes to hide old tablets specification pages so I can't compare this alongside a spec sheet at this time, is this the value you got when you measured it in `tablet > tablet debugger`?

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "MouseButtons": null,
+    "Touch": null
+  },
+  "DigitizerIdentifiers": [
+    {
+      "VendorID": 21827,
+      "ProductID": 77,
+      "InputReportLength": 8,
+      "OutputReportLength": 0,
+      "ReportParser": "OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenOffsetAuxReportParser",
```
Comment (jamesbt365) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:26]: Are you sure this is the right parser? Its currently only used on the UGEE M708 (UGEE and XP-Pen are close companies so its fine if this is the right parser)

The only difference is where it reads auxiliary data from, so if it reads fine this is the right parser.

**Review by jamesbt365 — CHANGES_REQUESTED** [review]
We also strongly prefer a diagnostic be provided as per our [CONTRIBUTING.md](https://github.com/OpenTabletDriver/OpenTabletDriver/blob/master/CONTRIBUTING.md) when adding support for tablets. Please provide the file generated from `help > export diagnostics`

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 1 above)
Comment (Mat931) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:30]: Yes. I tried removing every single one of them. 123 is to switch to the correct aux button format but that also seems to mess up the dpi. It sometimes works if the 100 is removed but not reliably. These were captured using usbmon from the Windows driver running inside a virtual machine.

Comment (Mat931) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:30]: How do I use the string reader?

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 2 above)
Comment (Mat931) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:7]: The MaxX and MaxY are from the tablet debugger. The width and height are copied from a different 16" tablet.

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 3 above)
Comment (Mat931) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:11]: MaxPressure is also the highest value seen in the tablet debugger

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 1 above)
Comment (jamesbt365) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:30]: The OEM driver will throw strings that the tablet does not need and are simply denied by the tablet.

put `21827`, `77` in the middle and ignore the bottom box, dump all.

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 4 above)
Comment (Mat931) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:26]: Yes. With other parsers the aux buttons don't work

**Comment by Mat931** [comment]
[string reader.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/14937913/string.reader.txt)
[Artist 16 Diagnostics.json](https://github.com/OpenTabletDriver/OpenTabletDriver/files/14937914/Artist.16.Diagnostics.json)

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 1 above)
Comment (jamesbt365) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:30]: Try only 100 and 123, replug your tablet to reset the initialization process.

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json (hunk 2 above)
Comment (jamesbt365) [reply OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 16.json:7]: You should calculate the values using the formula provided and use those instead.

**Comment by jamesbt365** [comment]
Functionality of this tablet will be spotty until #3245 is merged.

**Comment by Mat931** [comment]
[Artist 16 Diagnostics Win.json](https://github.com/OpenTabletDriver/OpenTabletDriver/files/14938647/Artist.16.Diagnostics.Win.json)

**Review by jamesbt365 — APPROVED** [review]

**Comment by InfinityGhost** [comment]
Blocking this PR until it has a backport

**Review by jamesbt365 — APPROVED** [review]

**Pull request merged**